strings.libsonnet

```
@@ -0,0 +1,127 @@
+local d = import 'doc-util/main.libsonnet';
+
+{
+'#': d.pkg(
+    name='strings',
+    url='github.com/jsonnet-libs/xtd/strings.libsonnet',
+    help=|||
+        `strings` implements few helpers functions to manipulate strings.
+    |||
+    ),
+
+    isUpcase(c)::    std.codepoint(c) >= 65 && std.codepoint(c) <= 90,   // between A & Z
+    isDowncase(c)::  std.codepoint(c) >= 97 &&  std.codepoint(c) <= 122, // between a & z
+    isDigit(c) ::    std.codepoint(c) >= 48 && std.codepoint(c) <= 57,   // between 0 & 9
+    isAlphaNum(c)::  self.isUpcase(c) || self.isDowncase(c) || self.isDigit(c),
+
+
+    '#camelcase':: d.fn(
+        |||
+            `camelcase` turns `str` string into its camelcase version.
+            By default, first letter will be capitalized, use `lower=true` to keep it down case.
+            E.g.
+            * `camelcase("A simple string")` → `"ASimpleString"`
+            * `camelcase("foo_bar",lower=true)` → `"fooBar"`
+            * `camelcase("+++More symbols!!!")` → `"MoreSymbols"` 
+
+            Function alias: `camelcase` 
+        |||,
+        [
+            d.arg('str', d.T.string),
+            d.arg('lower', d.T.boolean, default=false),
+        ]
+    ),
+    camelize(w,lower=false): self.camelcase(w, lower),
```

You can leave out the function arguments here, it gets inherited.

```suggestion
    camelize: self.camelcase,
```